tutorial/3_loading_data/data-pytorch-use-pytorch-dataset-function.ipynb

```diff
@@ -0,0 +1,457 @@
+    "visualize_sample_images(PATH_TO_DATASET)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "It might be worth noting that the image labels are listed in wnids.txt that can be found in the archive."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Training a classifier from this data\n",
+    "Now we have some understanding of what the database does and we are ready to do some ML on it.\n",
+    "\n",
+    "First we will define our machine learning model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "Using cache found in /cephyr/users/klim/Alvis/.cache/torch/hub/NVIDIA_DeepLearningExamples_torchhub\n",
+      "/cephyr/users/klim/Alvis/.cache/torch/hub/NVIDIA_DeepLearningExamples_torchhub/PyTorch/Classification/ConvNets/image_classification/models/common.py:13: UserWarning: pytorch_quantization module not found, quantization will not be available\n",
+      "  warnings.warn(\n",
+      "/cephyr/users/klim/Alvis/.cache/torch/hub/NVIDIA_DeepLearningExamples_torchhub/PyTorch/Classification/ConvNets/image_classification/models/efficientnet.py:17: UserWarning: pytorch_quantization module not found, quantization will not be available\n",
+      "  warnings.warn(\n"
+     ]
+    }
+   ],
+   "source": [
+    "# We will use torch.hub to load a pretrained model\n",
+    "efficientnet = torch.hub.load('NVIDIA/DeepLearningExamples:torchhub', 'nvidia_efficientnet_b0', pretrained=True)\n",
+    "#preprocessing_utils = torch.hub.load('NVIDIA/DeepLearningExamples:torchhub', 'nvidia_convnets_processing_utils')\n",
+    "\n",
+    "# We freeze all parameters except the last layer\n",
+    "freeze_blocks = [efficientnet.stem] + [\n",
+    "    layer for layer in efficientnet.layers if layer != efficientnet.classifier\n",
+    "]\n",
+    "for block in freeze_blocks:\n",
+    "    for parameter in block:\n",
+    "        parameter.requires_grad = False\n",
+    "\n",
+    "# Modify the number of output classes to 200\n",
+    "efficientnet.classifier.fc = nn.Linear(efficientnet.classifier.fc.in_features, 200)\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model = efficientnet\n",
+    "opt = optim.Adam(model.parameters(), lr=0.003)\n",
+    "loss_func = nn.CrossEntropyLoss()\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "- Initialize DataLoader using TinyImageNetDataset\n",
+    "- Constructs a DataLoader for the TinyImageNet training dataset.\n",
+    "- This will be used to iterate through the data in batches during training."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dataloader = get_dataloader(PATH_TO_DATASET)"
```

Comment:
```suggestion
    "from pytorch_dataset import TinyImageNetDataset\n"
    "dataset = TinyImageNetDataset(path_to_dataset)\n"
    "dataloader = DataLoader(dataset, ...)"
```
Perhaps something like this in 4 and 5. That way we expose the options to the DataLoader directly in the notebook.

Reply:
ok, let me check these sections code.